3rdparty/CMakeLists.txt

```diff
@@ -241,6 +245,32 @@ add_library(aesGladman OBJECT
 	aesGladman/sha1.c
 )
 
+# MacOS patches
+if(APPLE)
+    set(_NBL_MACOS_PATCH_INCLUDE_ "${CMAKE_CURRENT_BINARY_DIR}/macos/include" CACHE INTERNAL "")
+
+    set(_NBL_MACOS_BYTESWAP_H_
+"#if defined(__APPLE__)\n\
+// Mac OS X / Darwin features\n\
+#include <libkern/OSByteOrder.h>\n\n\
+#define bswap_16(x) OSSwapInt16(x)\n\
+#define bswap_32(x) OSSwapInt32(x)\n\
+#define bswap_64(x) OSSwapInt64(x)\n\
+#endif // __APPLE__"
+    )
+
+    set(_NBL_MACOS_ENDIAN_H_
+"#if defined(__APPLE__)\n\
+#include <machine/endian.h>\n\
+#endif // __APPLE__"
+    )
+
+    file(WRITE "${_NBL_MACOS_PATCH_INCLUDE_}/byteswap.h" "${_NBL_MACOS_BYTESWAP_H_}")
+    file(WRITE "${_NBL_MACOS_PATCH_INCLUDE_}/endian.h" "${_NBL_MACOS_ENDIAN_H_}")
+
+    include_directories(BEFORE SYSTEM "${_NBL_MACOS_PATCH_INCLUDE_}")
+endif()
```

Comment:
both ARM and x86 are little-endian

we're not going to support PowerPC cause its freaking ancient

what dependency even needs this?